src/variant_detection/variant_output.cpp

```diff
@@ -1,28 +1,104 @@
+                          .description = "Genotype"});
+
+    for (auto const & [id, length] : references_lengths)
+        hdr.contigs.push_back({ .id = id, .length = length});
+
+    // std::string filedate = "filedate=", transTime(),
+    hdr.other_lines = {"filedate=" + transTime(),
+                       "source=iGenVarCaller",
+                       "ALT=<ID=DEL,Number=1,Description=\"Deletion\">",
+                       "ALT=<ID=DUP:TANDEM,Number=1,Description=\"Tandem Duplication\">",
+                       "ALT=<ID=INS,Number=1,Description=\"Insertion of novel sequence\">"};
+    hdr.column_labels = {"CHROM", "POS", "ID", "REF", "ALT", "QUAL", "FILTER", "INFO", "FORMAT", sample_name};
+}
+
+/*! \brief Detects genomic variants from junction clusters and saves them in a record vector.
+ *
+ * \param[in] clusters     - input junction clusters
+ * \param[in] args         - command line arguments:\n
+ *                           **args.min_var_length** - minimum length of variants to detect
+ *                                                     (expected to be non-negative) - *default: 30 bp* \n
+ *                           **args.max_var_length** - maximum length of variants to detect
+ *                                                     (expected to be non-negative) - *default: 1,000,000 bp*\n
+ *                           **args.max_tol_inserted_length** - longest tolerated inserted sequence at non-INS SV types
+ *                                                              (expected to be non-negative) - *default: 5 bp*\n
+ *                           **args.min_qual** - minimum quality (amount of supporting reads) of a structural variant
+ *                                               (expected to be non-negative) - *default: 1 supporting read*\n
+ * \param[in, out] records - vector of SV records
+ *
+ * \details Extracts genomic variants from given junction clusters.
+ *          The quality of an SV is estimated based on the size of the cluster
+ *          (i.e. the number of reads supporting the SV).
+ */
+void write_records(std::vector<Cluster> const & clusters,
+                   cmd_arguments const & args,
+                   std::vector<bio::var_io::default_record<>> & records)
 {
-    variant_header header{};
-    header.set_fileformat("VCFv4.3");
-    header.add_meta_info("SVTYPE", 1, "String", "Type of SV called.", "iGenVarCaller", "1.0");
-    header.add_meta_info("SVLEN", 1, "Integer", "Length of SV called.", "iGenVarCaller", "1.0");
-    header.add_meta_info("END", 1, "Integer", "End position of SV called.", "iGenVarCaller", "1.0");
-    header.print(references_lengths, args.vcf_sample_name, out_stream);
     for (size_t i = 0; i < clusters.size(); ++i)
     {
         size_t cluster_size = clusters[i].get_cluster_size();
+        // ignore low quality SVs
         if (cluster_size >= args.min_qual)
         {
+            bio::var_io::default_record<> rec;
```

Comment:
If you write lots of records, it might be faster to create the record outside the loop and just overwrite the values inside.

Reply:
Then you also only need to change the fields that you actually update.

Reply:
[this only applies if you write the records directly, see below]